src/main/java/org/glite/authz/pep/pip/provider/policynamespip/PolicyNamesCache.java

```diff
@@ -0,0 +1,625 @@
+/*
+ * Copyright (c) Members of the EGEE Collaboration. 2006-2010.
+ * See http://www.eu-egee.org/partners/ for details on the copyright holders.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Authors:
+ * 2016-
+ *    Mischa Salle <msalle@nikhef.nl>
+ *    Rens Visser <rensv@nikhef.nl>
+ *    NIKHEF Amsterdam, the Netherlands
+ *    <grid-mw-security@nikhef.nl>
+ */
+
+package org.glite.authz.pep.pip.provider.policynamespip;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.HashSet;
+import java.nio.file.Path;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.attribute.FileTime;
+import java.nio.charset.Charset;
+import java.io.BufferedReader;
+import java.net.URLDecoder;
+import java.util.Calendar;
+
+import java.io.IOException;
+import java.text.ParseException;
+
+
+/**
+ * Subclass for {@link org.glite.authz.pep.pip.provider.PolicyNamesPIP}
+ * providing a parsed cache of the subjectdn entries in the info files in the
+ * trust directory.
+ * @author Mischa Sall&eacute;
+ */
+public class PolicyNamesCache {
+    /** Class logger instance */
+    private final Logger log = LoggerFactory.getLogger(PolicyNamesCache.class);
+
+
+    ////////////////////////////////////////////////////////////////////////
```

Comment:
I find it helpful, but we can remove them if you like. I left them in for the time being...